integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/rdd/SecondaryIndexCreator.scala

```diff
@@ -398,6 +404,28 @@ object SecondaryIndexCreator {
           secondaryIndexModel.sqlContext.sparkSession,
           carbonLoadModelForMergeDataFiles.getFactTimeStamp,
           rebuiltSegments)
+
+        if (isInsertOverwrite) {
+          var staleSegmentsList = new ListBuffer[String]()
+          SegmentStatusManager
+            .readLoadMetadata(indexCarbonTable.getMetadataPath).foreach { loadMetadata =>
+            if (!successSISegments.contains(loadMetadata.getLoadName)) {
```

Comment:
you can use directly filter instead of creating new buffer

Reply:
done